### PR TITLE
fixing and updating CI testing

### PR DESCRIPTION
**Pull request by DenysGonchar**
Switch from TravisCI to GH Actions

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -10,13 +10,17 @@ on:
   pull_request:
     branches: [ master ]
 
+  # Run this workflow every day
+  schedule:
+    - cron: '0 0 * * *'
+
```
Comment (chrzaszcz) [inline .github/workflows/CI.yml:16]: Why? Shouldn't it be triggered by updates?

File: ci_testing/run_test.sh [hunk 2]
```diff
@@ -4,5 +4,5 @@
 set -e
 
 rebar3 ct
-
 rebar3 dialyzer
+rebar3 xref
```
Comment (chrzaszcz) [inline ci_testing/run_test.sh]: Please add newline.

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -20,7 +20,7 @@ jobs:
     name: OTP ${{matrix.otp}}
     strategy:
       matrix:
-        otp: ['22.3.4', '21.3']
+        otp: ['22.3.4', '23.1']
```
Comment (chrzaszcz) [inline .github/workflows/CI.yml]: This looks so much like an accident, needs a comment in the commit message. Why is this changed twice, in two different commits?

File: rebar.config [hunk 4]
```diff
@@ -1,7 +1,17 @@
 {erl_opts, [debug_info]}.
 {deps, [{escalus, ".*", {git, "git://github.com/esl/escalus.git", {branch, "master"}}}]}.
 
-{relx, [{release, {mercutio, "0.0.1"},[mercutio, runtime_tools]},
+{relx, [{release, {mercutio, "0.0.1"},[
+           mercutio,
+           p1_utils,
```
Comment (chrzaszcz) [inline rebar.config:6]: Could you drop a note in the commit message why this is needed?

**Review by chrzaszcz — COMMENTED** [review]
Nice to see the progress. Just a few commits form me.

File: .github/workflows/CI.yml (hunk 3 above)
Comment (DenysGonchar) [reply .github/workflows/CI.yml]: It's not an accident but rather an iterative process:
* First I fixed testing as it was on Travis (with OTP 23.1).
* Then I did update of the dependencies
* Then i made it building and testing with OTP 23.1

Before `rebar.lock` update, it was not building on OTP 23.1 at all.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (DenysGonchar) [reply .github/workflows/CI.yml:16]: it should and will be triggered by updates.

but since we have `master` branch  version specified in `rebar.config` for escalus, that would be nice to have some regular check to ensure that escalus master version is not broken and still suitable for XMPP component development.  Note that `rebar3 upgrade` is one of the CI steps

File: rebar.config (hunk 4 above)
Comment (DenysGonchar) [reply rebar.config:6]: 8 months ago Nelson added `fast_tls` dependency to `escalus`, `fast_tls` in its turn has `p1_utils` dependency but doesn't list it in [*.app.src](https://github.com/processone/fast_tls/blob/master/src/fast_tls.app.src#L29) file so it's not automatically resolved, and we have to add it to relx configuration.

I've created an [issue](https://github.com/processone/fast_tls/issues/50) for fast_tls, also added note in the commit message

File: .github/workflows/CI.yml (hunk 3 above)
Comment (chrzaszcz) [reply .github/workflows/CI.yml]: Ok I see, it's just coincidence: `21.3` -> `23.1` look suspicious ;-)

**Review by chrzaszcz — APPROVED** [review]
Looks good, thanks for the additional improvements!
